plugins/pulp_rpm/plugins/importers/yum/upload.py

```diff
@@ -186,19 +186,18 @@ def _handle_erratum(repo, type_id, unit_key, metadata, file_path, conduit, confi
     unit_data.update(metadata or {})
     unit_data.update(unit_key or {})
 
-    existing_unit = model_class.objects.filter(**unit_key).first()
-    new_unit = model_class(**unit_data)
+    unit = model_class(**unit_data)
+    unit._repo_id = repo.repo_id
```

Comment:
I did some thinking about why this feels awkward, from a high level. If we were writing this from scratch (but still stuck with mongodb), is this how we would do it?

Here's what I came up with. The problem comes down to the fact that we're performing an action that makes changes with two different models, and we're trying to manage all of that from a behavior on just one of the models. This is a classic use case for a controller, and a good example of the awkwardness that ensues when one model's behavior depends on another model.

What if we keep the models as simple as we can, and make a controller function that knows how to create and update an Errata with its package lists? You could call it from the sync and upload code, and then naturally pass in a unit key, metadata, repo_id, and whatever else it might need.

Perhaps we didn't think of this sooner because we don't have any other examples of controllers in plugins.

Deletion seems like a case where having the Errata model auto-delete its package list entries is reasonable, similar to an automatic cascade delete. The only cross-model coupling being enforced would be the implicit coupling of the data model itself, aka the foreign key relationship. Nobody would lose sleep over that.

What do you think?